app/scripts/controllers/rif/rns/resolver/index.js

```diff
@@ -47,4 +50,42 @@ export default class RnsResolver extends RnsJsDelegate {
         }).catch(error => reject(error));
     });
   }
+  
+  getDomainDetails(domainName, address) {
+    return new Promise((resolve, reject) => {
+      this.checkIfSubdomainAndGetExpirationRemaining(domainName).then(remainingDays => { 
+        //Here i have the expiration in remainingDays
+        this.getOwner(domainName)
+        .then(ownerAddress => {
+          //Here i have the owner address
+          resolve(new DomainDetails('0x0', '0xabcd', remainingDays, false, ownerAddress));
+        }).catch(error => reject(error));        
+      }).catch(error => reject(error));
+    });
+  }
+
+  checkIfSubdomainAndGetExpirationRemaining(domainName) {  
```

Comment:
This method returns the remaining days if there is no error, so why not name it accordly?

Reply:
Oh! There in last commit i just renamed all to let the things more understandable, that was just a copy pasted function from the manager haha